shared/macros-ansible.jinja

```diff
@@ -504,3 +504,14 @@ See official documentation: https://jinja.palletsprojects.com/en/2.11.x/template
         state: present
   when: audit_syscalls_matched_{{{ arch }}}_audit_rules < audit_syscalls_number_of_syscalls
 {{%- endmacro %}}
+
+{{% macro ansible_ini_file_set(filename, section, key, value, description="") -%}}
+- name: "{{{ description if description else ("Set '" + key + "' to '" + value + "' in the [" + section + "] section of '" + filename + "'") }}}"
+  ini_file:
+    path: "{{{ filename }}}"
+    section: "{{{ section }}}"
+    option: "{{{ key }}}"
+    value: "{{{ value }}}"
+    create: yes
```

Comment:
So, if non-applicability applies in ansible remediation as well this should be `create: no`. Then remove `mode: 0600` to keep current permissions. Otherwise if we want to enforce the permissions I would go with `0644` instead of this too much restrict `0600`.

Reply:
This applicability in remediations is an extremely tricky business. In your PR, it's easier, as we know that we are dealing with a particular directory, whereas a generic (in this case `.ini`) macro may be applicable even if a file doesn't exist (consider e.g. `/etc/cron.d/my-job` and other `.d` directories).
Moreover, if we remove `create`, then IIRC the Ansible playbook will fail if the file doesn't exist. What we would like to see would be applicability expressed using the `when:` key, so the task is skipped.
So I would consider this case a rabbit hole to be dealt with later, when we improve the general applicability support for our content, and I would suggest to take the low-hanging fruits.

Reply:
Okay, it makes sense. Thanks for the explanation.